packages/@lwc/perf-benchmarks/src/utils/runJsFrameworkBenchmark.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { createElement } from '@lwc/engine-dom';
+import JsFrameworkBenchmarkTable from '@lwc/perf-benchmarks-components/dist/dom/benchmark/jsFrameworkBenchmarkTable/jsFrameworkBenchmarkTable.js';
+import { destroyComponent, insertComponent } from './utils.js';
+
+const actionsToElementIds = {
+    run: '#run',
+    runLots: '#runlots',
+    add: '#add',
+    update: '#update',
+    clear: '#clear',
+    swapRows: '#swaprows',
+};
+
+export const WARMUP_COUNT = 5;
+
+// This is an abstraction of the various operations that js-framework-benchmark does.
+// Most of these revolve around clicking an element with a given id (#run, #runlots, etc.), but sometimes
+// it involves clicking a particular element inside a row (e.g. to remove the row or select it).
+export function runJsFrameworkBenchmark(
+    name,
+    { benchmark, before, run, after },
+    { warmup, execute }
+) {
+    benchmark(name, () => {
+        let elm;
+        let controller;
+
+        before(async () => {
+            elm = createElement('benchmark-js-framework-benchmark-table', {
+                is: JsFrameworkBenchmarkTable,
+            });
+            await insertComponent(elm);
+
+            controller = {
+                // actions that only require clicking a given element ID
+                ...Object.fromEntries(
+                    Object.entries(actionsToElementIds).map(([action, id]) => {
+                        return [
+                            action,
+                            async () => {
+                                elm.querySelector(id).click();
```

Comment:
We _could_ extract out these `querySelector`s so that we don't pay the cost in the actual benchmark run, but 1) it doesn't seem to affect the numbers much, and 2) `js-framework-benchmark` seems to be implicitly doing this as well (assuming that WebDriver/Puppeteer delegate out to `document.querySelector`).